cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/CreateSubnetAction.java

```diff
@@ -0,0 +1,154 @@
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED),
+                    @Response(text = Outputs.FAILURE, field = Outputs.RETURN_CODE, value = Outputs.FAILURE_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR, isOnFail = true)
+            })
+    public Map<String, String> execute(@Param(value = ENDPOINT) String endpoint,
+                                       @Param(value = IDENTITY, required = true) String identity,
+                                       @Param(value = CREDENTIAL, required = true, encrypted = true) String credential,
+                                       @Param(value = PROXY_HOST) String proxyHost,
+                                       @Param(value = PROXY_PORT) String proxyPort,
+                                       @Param(value = PROXY_USERNAME) String proxyUsername,
+                                       @Param(value = PROXY_PASSWORD, encrypted = true) String proxyPassword,
+                                       @Param(value = HEADERS) String headers,
+                                       @Param(value = QUERY_PARAMS) String queryParams,
+                                       @Param(value = VERSION) String version,
+                                       @Param(value = AVAILABILITY_ZONE) String availabilityZone,
+                                       @Param(value = VPC_ID, required = true) String vpcId,
+                                       @Param(value = CIDR_BLOCK, required = true) String cidrBlock) {
+        try {
+            version = InputsUtil.getDefaultStringInput(version, "2016-09-15");
+
+            final CommonInputs commonInputs = new CommonInputs.Builder()
+                    .withEndpoint(endpoint)
+                    .withIdentity(identity)
+                    .withCredential(credential)
+                    .withProxyHost(proxyHost)
+                    .withProxyPort(proxyPort)
+                    .withProxyUsername(proxyUsername)
+                    .withProxyPassword(proxyPassword)
+                    .withHeaders(headers)
+                    .withQueryParams(queryParams)
+                    .withVersion(version)
+                    .withAction(CREATE_SUBNET)
+                    .withApiService(EC2_API)
+                    .withRequestUri(EMPTY)
+                    .withRequestPayload(EMPTY)
+                    .withHttpClientMethod(HTTP_CLIENT_METHOD_GET)
+                    .build();
+
+            final CustomInputs customInputs = new CustomInputs.Builder()
+                    .withAvailabilityZone(availabilityZone)
+                    .withVpcId(vpcId)
+                    .build();
+
+            final NetworkInputs networkInputs = new NetworkInputs.Builder().withCidrBlock(cidrBlock).build();
+
+            return new QueryApiExecutor().execute(commonInputs, customInputs, networkInputs);
```

Comment:
It would be nice if we would have a new output for the subnet id for a better UX.